### DISCUSSION TITLE
Official support request for booting error of HSR

### DISCUSSION THREAD

**Discussion by Yoo-Youngjae** [HumanSupportRobot/QandA#10] [Q&A]
Our HSR currently dose not boot at all.
After power on, its status indicator LED infinitely stucks at initial white light,
and its head display also stucks at initial screen.
Even we can not connect the internal computer through SSH.

We are sincerely looking for your help, because we are in the RoboCuP competition right now.
We hope to contact with your HSR support team in remote ASAP.

Our contact: dudwo235@snu.ac.kr

**Comment by yoshiaki-asahara** [comment]
I think HSR does not boot properly unless IP address was assigned to the HSR.
Does it correctly connect to the DHCP server provided by Robocup? Or is it set to the assigned IP address?
How about connecting your laptop with a wire and checking the network connection of the HSR?

**Reply by Yoo-Youngjae** [reply]
During Competition, we always set the robot to wired mode and connect it directly to the backpack laptop. Do you say that we should run a DHCP server at the laptop and give IP address to the robot?

**Reply by Yoo-Youngjae** [reply]
Now, connectting to the hsr administrator work well. But any rosnodes are not working.

**Reply by Yoo-Youngjae** [reply]
Can I get a diagnosis through real-time video meeting as soon as possible?

**Reply by yoshiaki-asahara** [reply]
Sorry i'm in other meeting now.
I think you couldn't see any nodes because roscore didn't work.
Would you try to set static IP on HSR if HSR doesn't connect to DHCP server?

**Reply by yoshiaki-asahara** [reply]
Would you also copy&paste the output of ifconfig here?

**Reply by Yoo-Youngjae** [reply]
here is "ifconfig"
![IMG_6221](https://user-images.githubusercontent.com/44061214/178399583-54ec5325-6559-499f-8b50-519103899ff7.JPG)
![IMG_6220](https://user-images.githubusercontent.com/44061214/178399603-99874c67-d425-4ee2-84b9-553d60f01df5.JPG)

here is "/etc/network/interfaces.wlan"
![IMG_6222](https://user-images.githubusercontent.com/44061214/178399646-292e367a-66b8-4658-b61b-0c9c8ca875e3.JPG)

And we confirmed that we can access inside and type "roscore" and turn on the roscore manually.

**Reply by Yoo-Youngjae** [reply]
And we set static ip now. But not standing too.

**Reply by yoshiaki-asahara** [reply]
Thank you for the images.
What happens if you turn off the HSR after setting the static IP and then boot the HSR as usual?

**Reply by Yoo-Youngjae** [reply]
1. After power on, its status indicator LED infinitely stucks at initial white light,
2.  its head display also stucks at initial screen.
3. connecting to the hsr administrator work well. But any rosnodes are not working.

**Reply by yoshiaki-asahara** [reply]
Thank you, I could understand the situation well.
Unfortunately, I can't think of any other solutions except IP addresses in my experience in this case.
I'm in the USA now and a little bit difficult to communicate with you for a long time due to the time difference.
Instead, I invited one of the engineers in Toyota.
Just a moment, please.

**Reply by hsr-admin** [reply]
Sorry, we do not offer videoconferencing services. We do not provide support for video conferencing.

The hsrb.robot.service will start the ros node by monitoring for the amplifier to be turned on.
The fact that the LED lights up white means that the amplifier power is turned on.
If the ros node is not started, it is likely that hsrb.robot.service has not started properly due to some problem.

Please check the status of the following services before releasing the stop button.
If it is normal, it is as follows.

`$ sudo systemctl status docker.hsrb.roscore.service
[sudo] administrator のパスワード:
● docker.hsrb.roscore.service - HSRB Core service
     Loaded: loaded (/etc/systemd/system/docker.hsrb.roscore.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2022-07-13 16:54:38 JST; 15min ago
    Process: 1771 ExecStartPre=/usr/bin/install -d -m755 -o ros -g ros /var/run/ros (code=exited, status=0/SUCCESS)
    Process: 1772 ExecStartPre=/usr/bin/install -d -m755 -o ros -g ros /var/run/ros/workspace (code=exited, status=0/SUCCESS)
    Process: 1773 ExecStartPre=/bin/bash -c . /etc/opt/tmc/robot/roshost.sh; /bin/systemctl set-environment $(get_roshost) (cod>
    Process: 2461 ExecStartPre=/usr/bin/docker rm -f docker.hsrb.roscore.service (code=exited, status=0/SUCCESS)
    Process: 2501 ExecStartPre=/bin/sh -c exec /usr/bin/docker run      --name docker.hsrb.roscore.service      --detach       >
   Main PID: 2857 (docker)
      Tasks: 11 (limit: 19035)
     Memory: 19.8M
     CGroup: /system.slice/docker.hsrb.roscore.service
             mq2857 /usr/bin/docker logs -f docker.hsrb.roscore.service

 7月 13 16:54:41 hsrb docker.hsrb.roscore.service[2857]: ROS_MASTER_URI=http://localhost:11311
 7月 13 16:54:41 hsrb docker.hsrb.roscore.service[2857]:
 7月 13 16:54:41 hsrb docker.hsrb.roscore.service[2857]: setting /run_id to 20220713-075440_0ca6beb2-0281-11ed-8f70-00306444d248
 7月 13 16:54:41 hsrb docker.hsrb.roscore.service[2857]: process[rosout-1]: started with pid [83]
 7月 13 16:54:41 hsrb docker.hsrb.roscore.service[2857]: started core service [/rosout]
 7月 13 16:54:41 hsrb docker.hsrb.roscore.service[2857]: process[hsrb/rosout_journald-2]: started with pid [86]
 7月 13 16:54:41 hsrb docker.hsrb.roscore.service[2857]: process[hsrb/diagnostics_journald-3]: started with pid [87]
 7月 13 16:54:41 hsrb docker.hsrb.roscore.service[2857]: process[hsrb/runtime_logger-4]: started with pid [88]
 7月 13 16:54:41 hsrb docker.hsrb.roscore.service[2857]: process[hsrb/diagnostic_aggregator-5]: started with pid [89]



~$ sudo systemctl status docker.hsrb.robot.service
● docker.hsrb.robot.service - HSRB Robot service
     Loaded: loaded (/etc/systemd/system/docker.hsrb.robot.service; enabled; vendor preset: enabled)
     Active: active (running) since Wed 2022-07-13 16:54:54 JST; 16min ago
    Process: 3901 ExecStartPre=/bin/rm -f ${ROS_HOME}/map (code=exited, status=0/SUCCESS)
    Process: 3902 ExecStartPre=/bin/bash -c . /etc/opt/tmc/robot/roshost.sh; /bin/systemctl set-environment $(get_roshost) (cod>
    Process: 3916 ExecStartPre=/usr/bin/docker rm -f docker.hsrb.robot.service (code=exited, status=0/SUCCESS)
    Process: 3936 ExecStartPre=/bin/sh -c exec /usr/bin/docker run      --name docker.hsrb.robot.service        --detach       >
   Main PID: 4010 (docker)
      Tasks: 11 (limit: 19035)
     Memory: 19.4M
     CGroup: /system.slice/docker.hsrb.robot.service
             mq4010 /usr/bin/docker logs -f docker.hsrb.robot.service`

To be sure, reset the roscore service as follows and try what happens when the stop button is deactivated.
$ sudo systemctl restart docker.hsrb.robot.service

If it does not start properly, you probably have some errors.
Use the journalct command to check the error.


(Reference)
　https://docs.hsr.io/hsr_log_analysis_manual_en/log_files.html#journald

**Reply by hsr-admin** [reply]
You will need to have an internet connection, but you can use the "hsrb_command upload_log" command to collect the logs and I will check the logs for you.
Reference)
　https://docs.hsr.io/hsrb_user_manual_en/troubleshoot/troubleshoot.html#upload-log-tool

**Reply by Yoo-Youngjae** [reply]
We solved it. It was network setting problem. If we have any more problem, we'll make a issue again.
Thank you for your help :)

**Reply by hsr-admin** [reply]
I am glad the problem has been resolved.
Can you please write the cause of the problem and how to deal with it to share the information, which can be done after the convention is over.
